### ISSUE TITLE
[TW-670] Trouble Editing Task Due Date

### ISSUE THREAD

**Issue by taskwarrior** [GothenburgBitFactory/taskwarrior#705]
_Michelle Crane on 2010-07-27T10:49:01Z says:_ 

 I am now able to use nano to edit my tasks (windows xp/cygwin installation).  See http://taskwarrior.org/boards/1/topics/show/590 to how-to.

I want my due dates to be specific to my working day (i.e., I don't want my tasks due at midnight on the day they are due, rather at 4 p.m.  To enable this, my .taskrc has the following configuration:

<pre>
dateformat=Y-M-D.H:N
dateformat.annotation=[Y-M-D H:N]
</pre>

The period in the dateformat is because I can't seem to make it work with a space.  And I need the H:N so I can add a new task with a deadline that includes a good time for me.

On the other hand, I like to read my annotations with enclosing [] and it doesn't force me to use a ., so I don't.

Now my problem is when I try to edit an existing task (say a recurring one) that has an old due date, i.e., the timing hasn't been changed from 00:00.  Here's what I see at my command-line:

<pre>
F:\Dropbox\My Dropbox\TextFiles>t info 43
Name        Value
* ---------- ------------------------------------
ID          43
Description test
Status      Pending
Due         2010-07-27.00:00
UUID        6d8a017f-49f4-27db-da7f-9510fa650196
Entered     2010-07-27.10:08 (29 secs)
F:\Dropbox\My Dropbox\TextFiles>t edit 43
Launching 'nano "/cygdrive/f/dropbox/My Dropbox/TextFiles/.task/task.4716.43.task"' now...
Editing complete.
Edits were detected.
F:\Dropbox\My Dropbox\TextFiles>t info 43
Name        Value
* ---------- ------------------------------------
ID          43
Description test
Status      Pending
Due         2010-07-27.00:00
UUID        6d8a017f-49f4-27db-da7f-9510fa650196
Entered     2010-07-27.10:08 (1 min)
</pre>

In the editing part, I did the following:

* change the time from 00:00 to 16:00

If I just did this, I would get an error:

<pre>
Editing complete.
Edits were detected.
Error: "[2010-07-27" is not a valid date (DEFAULT).
Task couldn't handle your edits.  Would you like to try again? (y/n)
</pre>

It seems that the simple act of editing the task creates an annotation (seen at the bottom of the editing window), in my default annotation format.  If I leave it as is, I get this error.  So, I also

* change the newly added annotation to use a . in the format. 

Then, my edit of the hours seems to stick, because it says

<pre>
Editing complete.
Edits were detected.
</pre>

But, it doesn't actually stick, as you can see from the trace above, because the hour hasn't changed at all.

So, something is ignoring the fact that I've edited the time of my task.

**Comment by taskwarrior** [comment]
Migrated metadata:
```
Created: 2010-07-27T10:49:01Z
Modified: 2014-02-09T02:06:32Z
```

**Comment by taskwarrior** [comment]
_Paul Beckingham on 2012-02-28T01:14:12Z says:_ 

 Fixed along with #942.
